pkg/controller/endpointslicemirroring/reconciler_test.go

```diff
@@ -791,6 +791,116 @@ func TestReconcile(t *testing.T) {
 		expectedClientActions:  2,
 		maxEndpointsPerSubset:  2,
 		expectedMetrics:        &expectedMetrics{desiredSlices: 2, actualSlices: 2, desiredEndpoints: 4, addedPerSync: 4, updatedPerSync: 0, removedPerSync: 0, skippedPerSync: 1, numCreated: 2, numUpdated: 0},
+	}, {
+		testName: "Annotation mirroring for endpoints with 2 subsets, multiple ports and addresses, existing EndpointSlice identical to subset",
+		subsets: []corev1.EndpointSubset{{
+			Ports: []corev1.EndpointPort{{
+				Name:     "http",
+				Port:     80,
+				Protocol: corev1.ProtocolTCP,
+			}, {
+				Name:     "https",
+				Port:     443,
+				Protocol: corev1.ProtocolUDP,
+			}},
+			Addresses: []corev1.EndpointAddress{{
+				IP:       "10.0.0.1",
+				Hostname: "pod-1",
+				NodeName: utilpointer.StringPtr("node-1"),
+			}, {
+				IP:       "10.0.0.2",
+				Hostname: "pod-2",
+				NodeName: utilpointer.StringPtr("node-2"),
+			}},
+		}, {
+			Ports: []corev1.EndpointPort{{
+				Name:     "http",
+				Port:     3000,
+				Protocol: corev1.ProtocolTCP,
+			}, {
+				Name:     "https",
+				Port:     3001,
+				Protocol: corev1.ProtocolUDP,
+			}},
+			Addresses: []corev1.EndpointAddress{{
+				IP:       "10.0.1.1",
+				Hostname: "pod-11",
+				NodeName: utilpointer.StringPtr("node-1"),
+			}, {
+				IP:       "10.0.1.2",
+				Hostname: "pod-12",
+				NodeName: utilpointer.StringPtr("node-2"),
+			}, {
+				IP:       "10.0.1.3",
+				Hostname: "pod-13",
+				NodeName: utilpointer.StringPtr("node-3"),
+			}},
+		}},
+		existingEndpointSlices: []*discovery.EndpointSlice{{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-ep-1",
+				Annotations: map[string]string{
+					corev1.LastAppliedConfigAnnotation: "{\"apiVersion\":\"v1\",\"kind\":\"Endpoints\",\"metadata\":{\"annotations\":{},\"name\":\"ep-test\",\"namespace\":\"default\"},\"subsets\":[{\"addresses\":[{\"ip\":\"193.99.144.80\"}],\"notReadyAddresses\":[{\"ip\":\"195.54.164.39\"}],\"ports\":[{\"name\":\"http\",\"port\":80,\"protocol\":\"TCP\"},{\"name\":\"https\",\"port\":443,\"protocol\":\"TCP\"}]}]}",
```

Comment:
I'm not sure this is the primary issue you want to be testing here. If I'm understanding this correctly, it ensures that an EndpointSlice that has this annotation will have it removed after a mirroring sync has occurred. That's helpful for cleanup, but what I think we actually need to test is that an Endpoints resource with this annotation is not translated to a mirrored EndpointSlice resource.

Reply:
agreed

Reply:
so, let's test:
1. endpoint with corev1.LastAppliedConfigAnnotation: "_data_here" is notmirrored
2. endpoint with corev1.LastAppliedConfigAnnotation: empty is notmirrored
3. endpoint with 2 annotations:  corev1.LastAppliedConfigAnnotation: "_data_here" and foo:bar only mirrors foo bar

Reply:
Sounds good